python/mxnet/ndarray.py

```diff
@@ -878,6 +893,137 @@ def equal(lhs, rhs):
         None)
     # pylint: enable= no-member, protected-access
 
+def not_equal(lhs, rhs):
+    """Return (lhs != rhs) element-wise.
+
+    Parameters
+    ----------
+    lhs : Array or float value
+        left hand side operand
+
+    rhs : Array of float value
+        right hand side operand
+
+    Returns
+    -------
+    out: Array
+        result array
+    """
+    # pylint: disable= no-member, protected-access
+    return _ufunc_helper(
+        lhs,
+        rhs,
+        _internal._not_equal,
+        lambda x, y: 1 if x != y else 0,
+        _internal._not_equal_scalar,
+        None)
+    # pylint: enable= no-member, protected-access
+
+def greater(lhs, rhs):
+    """Return (lhs > rhs) element-wise.
+
+    Parameters
+    ----------
+    lhs : Array or float value
+        left hand side operand
+
+    rhs : Array of float value
+        right hand side operand
+
+    Returns
+    -------
+    out: Array
+        result array
+    """
+    # pylint: disable= no-member, protected-access
+    return _ufunc_helper(
+        lhs,
+        rhs,
+        _internal._greater,
+        lambda x, y: 1 if x > y else 0,
+        _internal._greater_scalar,
+        None)
```

Comment:
@piiswrong  The RHS and LHS scalar operators are not a problem after all. Since Python does the right thing  . We implement only the first argument as an NDArray(or Symbol). i.e __gt__(self, other). When a comparison operator is called with the first argument as a scalar i.e 1 > x( x being NDArray), Python appropriately calls less than operator, more precisely __lt__(self, other).
I have added the test cases though to cover this. Here is the pull request #4229